src/applications/inter-app-communication.md

```diff
@@ -0,0 +1,46 @@
+## Inter-app Communication
+
+### DNS Service Discovery
+
+Deis Workflow supports deploying a single app composed of a system of processes. Each Deis Workflow app communicates on a single port, so communicating with another Workflow app means finding that app's address and port. All Workflow apps are mapped to port 80 externally, so finding its IP address is the only challenge. Workflow creates a [Kubernetes Service](https://kubernetes.io/docs/user-guide/services/) for each app, which effectively assigns a name and one cluster-internal IP address to an app. The DNS service running in the cluster adds and removes DNS records which point from the app name to its IP address as services are added and removed. Deis Workflow apps, then, can simply send requests to the domain name given to the service, which is "app-name.app-namespace".
+
+For example, if you deployed a Deis Workflow app like this:
+
+    $ deis create http-server --no-remote
+    $ deis pull deis/example-dockerfile-python:latest -a http-server
+
+You can deploy a second app and communicate with the first one:
+
+    $ deis create http-client --no-remote
+    $ # Need to set a port because the docker image doesn't use one.
+    $ deis config:set PORT=8080 -a http-client
+    $ deis config:set HOSTNAME=http-server.http-server
+    $ deis pull -a http-client willfarrell/ping:latest
+    $ deis logs -a http-client
+    ...
+    --- http-server.http-server ping statistics ---
+    1 packets transmitted, 0 packets received, 100% packet loss
+    ping http-server.http-server every 300 sec
+    PING http-server.http-server (10.71.241.103): 56 data bytes
+    
+    $ # While the ping failed, it successfully resolved
+    $ #  "http-server.http-server" to "10.71.241.103"
+
+Clean up:
+
+    $ deis destroy -a http-server --confirm=http-server
```

Comment:
I don't believe cleanup should be documented. The user following the guide may want to play around with the example some more.